Makefile

```
@@ -116,9 +117,12 @@ O_FILES       := $(foreach f,$(S_FILES:.s=.o),build/$f) \
                  $(foreach f,$(C_FILES:.c=.o),build/$f) \
                  $(foreach f,$(wildcard baserom/*),build/$f.o)
 
+OVL_RELOC_FILES := $(shell grep -o '[^"]*_reloc.o' spec )
```

Shouldn't this read from build/spec (`build/$(SPEC)`)?

---

Ideally it should, but it can't read from the built spec since this is variable is resolved (and needed) before the spec is preprocessed as part of the make run

---

What makes this needed before the spec has a chance to preprocess? `DEP_FILES` usage?

Maybe a middle ground solution could be
```suggestion
OVL_RELOC_FILES := $(shell $(CPP) $(CPPFLAGS) spec | grep -o '[^"]*_reloc.o' )
```
if I got my shell-ing right

---

Would it be worth making a new variable for this if we want to do the same thing twice?

---

A variable for holding the preprocessed spec or the command to preprocess the spec?

---

I'm not sure, I'm just concerned about the duplication of
```make
OVL_RELOC_FILES := $(shell $(CPP) $(CPPFLAGS) spec | grep -o '[^"]*_reloc.o' )
[...]
build/$(SPEC): $(SPEC)
	$(CPP) $(CPPFLAGS) $< > $@
```

Maybe `PROCESS_SPEC = $(CPP) $(CPPFLAGS) $(SPEC)`?